packages/flutter_tools/lib/src/commands/run.dart

```diff
@@ -288,11 +292,32 @@ class RunCommand extends RunCommandBase {
       );
     }
 
+    DateTime appStartedTime;
+    // Sync completer so the completing agent attaching to the resident doesn't 
+    // need to know about analytics. 
+    //
+    // Do not add more operations to the future.
+    final Completer<Null> appStartedTimeRecorder = new Completer<Null>.sync();
+    appStartedTimeRecorder.future.then(
+      (Null _) { appStartedTime = clock.now(); }
```

Comment:
oops, our comments crossed paths midair. 

I didn't really understand the references to Future in the Completer's dartdoc either. It seems to be written for the perspective of the Completer's reader rather than the Completer's supplier. 

I'm trying to supply a Completer such that a piece of code runs synchronously when the owner of the Completer, a few layers deeper, calls complete() on it.